book/src/programmability/bcs.md

```diff
@@ -52,6 +52,10 @@ Structs encode similarly to simple types. Here is how to encode a struct using B
 {{#include ../../../packages/samples/sources/programmability/bcs.move:encode_struct}}
 ```
 
+Note that encoding a custom struct does not make much sense in interacting with the Move VM, as
```

Comment:
I'm not sure if this note is in place, there's no function available to developer which would deconstruct a struct from BCS.

Reply:
There is not a single function but I just tested that something like this is indeed working:
```
    const coinDecoder = bcs.struct(
        '0x2::coin::Coin<0x2::sui::SUI>',
        {
            id: bcs.struct('0x2::object::UID', {
                id: bcs.struct('0x2::object::ID', {
                    bytes: bcs.Address
                }),
            }),
            balance: bcs.struct('0x2::balance::Balance<0x2::sui::SUI>', {
                value: bcs.u64()
            })
        }
    );

    for (const coin of coinsResp.data) {
        if (coin.data?.bcs?.dataType !== 'moveObject') {
            continue;
        }
        const data = fromB64(coin.data.bcs.bcsBytes);
        const coinData = coinDecoder.parse(data);
        console.log(coinData);
    }
```
What I am referring to here, is that while doing `bcs.struct` for the `parse` use-case has value, using it for `toBytes` use-case has none. Maybe this is not so clear currently?

Reply:
Well, `toBytes` or its Move equivalent `bcs::to_bytes` has a number of applications. Main one - encoding transaction inputs / data for bridge-like applications. I suggest removing these edits from the scope of this PR, and focus on the corrections for the transaction inputs part.

Reply:
Resolved in [5faa5e2](https://github.com/MystenLabs/move-book/pull/91/commits/5faa5e2e5dd18255df69c4260ca8370373a427c7)